### PR TITLE
[DEV-2527] Update critical data info cleaning operation validation

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR updates cleaning operation validation by
* trying to cast the imputed value to the assign column type
* checking whether the cleaning operation is supported by the column type

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/node/cleaning_operation.py
Comment (YS-L) [inline featurebyte/query_graph/node/cleaning_operation.py:306]: This set of dtypes seem common and also used in `DisguisedValueImputation`. Would it make sense to define it once and reuse?

File: tests/unit/query_graph/model/test_column_info.py
Comment (YS-L) [inline tests/unit/query_graph/model/test_column_info.py:37]: It is expected for the timezone information to be dropped in this case? Seems a bit strange to allow specifying a tz-aware CDI value for a tz-naive column.

Comment (gee-senbong) [reply tests/unit/query_graph/model/test_column_info.py:37]: We have similar case when casting float number to int (where the number after decimal is truncated). Thinking whether it will be a bit inconsistent if we block this case but ok for another case.

Comment (YS-L) [reply tests/unit/query_graph/model/test_column_info.py:37]: For timezone we might want to be more careful since dropping timezone information loses much more information than dropping decimals. If the timestamps in the tz-naive column are in UTC, I think a more reasonable casting process is to convert the tz-aware timestamp to UTC first before dropping the timezone.

Comment (gee-senbong) [reply tests/unit/query_graph/model/test_column_info.py:37]: That sounds to be a better solution than failing it. Will update the conversion logic here.

**Review by YS-L — APPROVED** [review]

**Comment by gee-senbong** [comment]
Will wait for (https://github.com/featurebyte/featurebyte/pull/1909) to get merged first before merging this PR.

**Comment by gee-senbong** [comment]
Tested this branch locally and it should not introduce additional backward compatibility of existing branch. Will merge this PR first as it is not clear when https://github.com/featurebyte/featurebyte/pull/1909 get merged.

**Pull request merged**
